src/boot/efi/stub.c

```diff
@@ -394,9 +394,11 @@ static EFI_STATUS run(EFI_HANDLE image) {
             !FLAGS_SET(loader_features, EFI_LOADER_FEATURE_RANDOM_SEED)) {
                 _cleanup_(file_closep) EFI_FILE *esp_dir = NULL;
 
-                err = partition_open(MAKE_GUID_PTR(ESP), loaded_image->DeviceHandle, NULL, &esp_dir);
-                if (err == EFI_SUCCESS) /* Non-fatal on failure, so that we still boot without it. */
-                        (void) process_random_seed(esp_dir);
+                if (loaded_image->DeviceHandle) {
```

Comment:
Please merge this with the above conditions `efivar_get_uint64_le(...)` and `!FLAGS_SET(...)`.